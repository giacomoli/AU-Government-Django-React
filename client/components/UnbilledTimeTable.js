import React from 'react'
import moment from 'moment'

import { formatCurrency } from 'utils'
// import Pre from 'components/Pre'

export default ({ unbilledTime, modal }) => (
  <table id="mainTable" className="table table-bordered table-striped m-b-0">
    <thead>
      <tr>
        <th>Date created</th>
        <th>Staff Member</th>
        <th>Description</th>
        <th>Billable</th>
        <th>Actual units</th>
        <th>Units to bill</th>
        <th>Rate</th>
        <th>Value</th>
      </tr>
    </thead>
    <tbody>
      {unbilledTime.map(time => (
        <tr key={`${time.__typename}-${time.id}`} onClick={() => modal(time)}>
          <td style={{ width: '10%' }}>{moment(time.date).format('DD/MM/YYYY')} {time.recordType === 2 && <span className="badge badge-info">sales</span>}</td>
          <td>{time.staffMember && time.staffMember.fullName }</td>
          <td style={{ whiteSpace: 'pre-line' }}>{time.description}</td>
          <td>{time.statusDisplay && time.statusDisplay}</td>
          <td>{time.units}</td>
          <td>{time.statusDisplay === 'Non billable' ? 0 : time.unitsToBill}</td>
          <td>{formatCurrency(time.rate)}</td>
          <td>{formatCurrency(time.cost)}</td>
        </tr>
      ))}
    </tbody>
    <style jsx>{`
      .delete {
        width: 10px;
        text-align: center;
      }
    `}</style>
  </table>
)
