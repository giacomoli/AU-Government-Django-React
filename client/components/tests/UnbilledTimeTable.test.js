import React from 'react'
import renderer from 'react-test-renderer'
import { ApolloProvider } from 'react-apollo'
import { reducer } from 'redux-form'
import { ApolloClient } from 'apollo-client'
import { createStore } from 'redux'
import moment from 'moment'
import UnbilledTimeTable from '../UnbilledTimeTable'

/* eslint-disable */
describe('UnbilledTimeTable', () => {
  const client = new ApolloClient()
  const store = createStore(reducer)
  const unbilledTime = {
    unbilledTime: [
      {
        __typename: "TymeEntryType",
        id: 12312,
        date: moment(),
        staffMember: {
          fullName: 'qwe qwe',
        },
        statusDisplay: 'Non billable',
        units: 2,
        unitsToBill: 12,
        rate: 12,
        cost: 12,
      },
      {
        __typename: "TymeEntryType",
        id: 12123312,
        date: moment(),
        staffMember: {
          fullName: 'qwe qwe',
        },
        statusDisplay: 'Billable',
        units: 2,
        unitsToBill: 12,
        rate: 12,
        cost: 12,
      }
  ]
  }

  it('should render UnbilledTimeTable', () => {
    const wrapper = renderer.create(
      <ApolloProvider client={client} store={store}>
        <UnbilledTimeTable
          unbilledTime={unbilledTime.unbilledTime}
        />
      </ApolloProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
