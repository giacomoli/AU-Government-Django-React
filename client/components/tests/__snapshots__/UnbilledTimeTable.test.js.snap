// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnbilledTimeTable should render UnbilledTimeTable 1`] = `
<table
  className="jsx-3598123328 table table-bordered table-striped m-b-0"
  id="mainTable"
>
  <thead
    className="jsx-3598123328"
  >
    <tr
      className="jsx-3598123328"
    >
      <th
        className="jsx-3598123328"
      >
        Date created
      </th>
      <th
        className="jsx-3598123328"
      >
        Staff Member
      </th>
      <th
        className="jsx-3598123328"
      >
        Description
      </th>
      <th
        className="jsx-3598123328"
      >
        Billable
      </th>
      <th
        className="jsx-3598123328"
      >
        Actual units
      </th>
      <th
        className="jsx-3598123328"
      >
        Units to bill
      </th>
      <th
        className="jsx-3598123328"
      >
        Rate
      </th>
      <th
        className="jsx-3598123328"
      >
        Value
      </th>
    </tr>
  </thead>
  <tbody
    className="jsx-3598123328"
  >
    <tr
      className="jsx-3598123328"
      onClick={[Function]}
    >
      <td
        className="jsx-3598123328"
        style={
          Object {
            "width": "10%",
          }
        }
      >
        16/01/2019
         
      </td>
      <td
        className="jsx-3598123328"
      >
        qwe qwe
      </td>
      <td
        className="jsx-3598123328"
        style={
          Object {
            "whiteSpace": "pre-line",
          }
        }
      />
      <td
        className="jsx-3598123328"
      >
        Non billable
      </td>
      <td
        className="jsx-3598123328"
      >
        2
      </td>
      <td
        className="jsx-3598123328"
      >
        0
      </td>
      <td
        className="jsx-3598123328"
      >
        $12.00
      </td>
      <td
        className="jsx-3598123328"
      >
        $12.00
      </td>
    </tr>
    <tr
      className="jsx-3598123328"
      onClick={[Function]}
    >
      <td
        className="jsx-3598123328"
        style={
          Object {
            "width": "10%",
          }
        }
      >
        16/01/2019
         
      </td>
      <td
        className="jsx-3598123328"
      >
        qwe qwe
      </td>
      <td
        className="jsx-3598123328"
        style={
          Object {
            "whiteSpace": "pre-line",
          }
        }
      />
      <td
        className="jsx-3598123328"
      >
        Billable
      </td>
      <td
        className="jsx-3598123328"
      >
        2
      </td>
      <td
        className="jsx-3598123328"
      >
        12
      </td>
      <td
        className="jsx-3598123328"
      >
        $12.00
      </td>
      <td
        className="jsx-3598123328"
      >
        $12.00
      </td>
    </tr>
  </tbody>
</table>
`;
